fedimint-core/src/task/jit.rs

```diff
@@ -114,70 +115,30 @@ where
 
         Self {
             _cancel_tx: cancel_tx,
-            val_rx: Arc::new(Some(val_rx).into()),
-            val: sync::OnceCell::new().into(),
+            shared: shared2,
         }
     }
 
     /// Get the reference to the value, potentially blocking for the
     /// initialization future to complete
-    pub async fn get_try(&self) -> std::result::Result<&T, OneTimeError<E>> {
-        /// Temporarily taken data out of `arc`, that will be put back on drop,
-        /// unless canceled
-        ///
-        /// Used to achieve cancelation safety.
-        struct PutBack<'a, T> {
-            val: Option<T>,
-            arc: &'a Arc<std::sync::Mutex<Option<T>>>,
-        }
-
```

Comment:
Avoiding the cancel safety complexity here seems like a good trade-off even though we get more restrictions on `T`.